fs/readdircount_linux_amd64.go

```diff
@@ -0,0 +1,67 @@
+// +build linux,amd64
+
+package fs
+
+import (
+	"errors"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func countDirEntries(buf []byte, n int) int {
+	count := 0
+	buf = buf[:n]
+	for len(buf) > 0 {
+		// see man page getdents(2) for struct linux_dirent64
+		reclen := *(*uint16)(unsafe.Pointer(&buf[16]))
+		ino := *(*uint64)(unsafe.Pointer(&buf[0]))
+
+		if int(reclen) > len(buf) {
+			return count
+		}
+		buf = buf[reclen:]
+		if ino == 0 {
```

Comment:
No, `.` and `..` should have real inodes numbers. AFAIU the case `ino == 0` could happen when there is no files on that directory entry (there is a hole in the record). I don't think that procfs returns such entries, but `ReadDirCount()` is supposed to be generic. I took that from:
https://golang.org/src/syscall/dirent.go?s=2134:2232#L78
```
    78		if ino == 0 { // File absent in directory.
    79			continue
    80		}
    81	
```